### PR TITLE
[MINT-795] Update github actions.

### PR DESCRIPTION
**Pull request by fendy-myeatclub**
We're updating the github workflow. 
Other jobs such as releasing new versions and pushing latest to ECR  will require lint and test to pass so I've consolidated the two existing files into one to access them.
The new work flow:
On every push we:

1. Run the linter
2. Run tests
3. build Images
4. Tag them with the timestamp, commit id, and branch name. To use branch name, I had to pipe it through various commands to normalize so that it docker tag complaint. So if you plan to use your branch as an `mjmlRef` in `sandbox_create` for testing, it is important that you look at the generated branch name in action because it maybe different from your actual branch name.
5. Login to ECR and push the image.

If the branch is master and previous jobs pass then:

1. We release new versions. Bump scheme is set to major.
2. We tag the image with both 'latest' and the version tag. 
3. We push the new tags to ECR.

This creates a separation of concern. 
1. MJML build and release is still automated but also contained to only the mjml-server repo.  
2. We can make changes to mjml in isolation.
3. We can share these version tags between both core and direct similar to ec payment even though we haven't migrated to ec payment in web lol.
4. We can now treat MJML as any other third party image in the scope of web and direct-backend. We can hardcode the image version used.
5. We can easily test, deploy and/or revert from different versions of mjml in web. 

Changes also included:

Updating `.gitignore`
Updating README

### CODE REVIEW

File: .github/workflows/build_and_release_ecr_image.yml
Comment (llostris) [inline .github/workflows/build_and_release_ecr_image.yml]: Let's not use master versions of actions; this assumes a huge trust in the maintainer and can lead to us unexpectedly upgrading the action – thus breaking our pipeline :)

That's also the recommendation of the author: https://github.com/eatclub/direct-backend/blob/4462cb44578ee263ed1481af28b5cf4d03491830/.github/workflows/deploy.yml#L71-L75

File: .github/workflows/build_and_release_ecr_image.yml
Comment (llostris) [inline .github/workflows/build_and_release_ecr_image.yml]:
```suggestion
name: CI & CD
```
for brevity :)

File: .github/workflows/build_and_release_ecr_image.yml
Comment (llostris) [inline .github/workflows/build_and_release_ecr_image.yml:31]: Minor, but we seem to be inconsistent with naming, and we use underscores in some, and then dashes in others :) 

Screenshot 2024-03-27 at 10 31 43

We can also add `name` to each job to add human-readable name, eg. like here: https://github.com/eatclub/direct-frontend/blob/main/.github/workflows/deploy.yml

File: README.md
Comment (michaelchu) [inline README.md]: Minor misspelling? 

Do you mean to write  `i.e.` instead of `i.ie`

**Review by michaelchu — APPROVED** [review]
One small comment, otherwise LGTM

File: README.md
Comment (llostris) [inline README.md]:
```suggestion
When testing in a sandbox, you can use the commit id, timestamp or branch to test. If you want to use branch
```

File: README.md
Comment (llostris) [inline README.md:101]: Nice 👍

File: .github/workflows/build_and_release_ecr_image.yml
Comment (llostris) [inline .github/workflows/build_and_release_ecr_image.yml:51]: A bit of future thinking: Direct will want to use this image too, both in prod, UAT and local dev :) Not immediately though, I'd expect we get to our ticket there in 2-3 weeks.

@LBarovic @lceatclub We might need your help with configuring cross-account access to this ECR image 🤔 Or at least some tips/pointing in the right direction

File: .github/workflows/build_and_release_ecr_image.yml
Comment (llostris) [inline .github/workflows/build_and_release_ecr_image.yml]: Screenshot 2024-03-28 at 10 25 05

Two open questions:
1. Do we want to auto-push the major version every time? 🤔 (Instead of patch/minor)
2. Should we clean up all the test tags? :) 

I was reading that the action allows to tweak per-PR the version by labeling `release:major`, `release:minor`, `release:patch`: https://github.com/rymndhng/release-on-push-action?tab=readme-ov-file#how-do-i-change-the-bump-version-scheme-using-pull-requests. But I think our default should be what we want by default/what in most cases would be used.

File: README.md
Comment (llostris) [inline README.md:21]: Add note here explaining:

1. After merging the PR, a new version will be automatically released and pushed to ECR with <X> version bump by default
2. Explain that engineers can choose to override release bump type byusing labels: release:major, release:minor, release:patch, norelease on their PRs :)

Comment (fendy-myeatclub) [reply README.md:21]: Do we want to include the ECR images for local dev or do we keep them  pointed to the other one? Biggest caveat to switching locally is they need to be logged in to pull from ECR.

File: .github/workflows/build_and_release_ecr_image.yml
Comment (fendy-myeatclub) [reply .github/workflows/build_and_release_ecr_image.yml]: Now using `0.28.0`, does that resolve the issue?

File: README.md
Comment (fendy-myeatclub) [reply README.md:21]: I'm also wondering if we should change the prefix from `v` to something like `mjml-v` just to make it easier to search and replace though thats not conventional. Thoughts?

Comment (llostris) [reply README.md:21]: I think I'd stick to convention for `v` 🤔 

At some point we might have to switch them anyway, so if anything – it wouldn't be a question of "if" or "when" (the old image might break, or get out of sync)

**Comment by fendy-myeatclub** [comment]
For context of new commits, by default Github action uses AMD64 which broke the container since we're expecting ARM64. Made changes to support ARM64.

File: .github/workflows/build_and_release_ecr_image.yml
Comment (llostris) [inline .github/workflows/build_and_release_ecr_image.yml]:
```suggestion
        uses: actions/checkout@v4
```
That's the up-to-date action :)

File: .github/workflows/build_and_release_ecr_image.yml
Comment (llostris) [inline .github/workflows/build_and_release_ecr_image.yml:18]:
```suggestion
        uses: actions/setup-node@v4
```

File: .github/workflows/build_and_release_ecr_image.yml
Comment (llostris) [inline .github/workflows/build_and_release_ecr_image.yml]:
```suggestion
        uses: actions/checkout@v4
```

**Comment by fendy-myeatclub** [comment]
@llostris Also note that we're not pushing logs to sumo logic since its being streamed and not written. Do you want to include that in the node upgrade or should it be a separate ticket?

File: .github/workflows/build_and_release_ecr_image.yml
Comment (LBarovic) [reply .github/workflows/build_and_release_ecr_image.yml:51]: No problem.  I added cross-account access from "direct-prod" and "direct-test" just now.  In the future, we may want to consider moving this out of the "core-prod" ECR registry.  Keeping it there is the easiest solution for now, but I'd like to consider creating a shared ECR registry in the "shared-infra-prod" account if we get any other images that need to be shared across multiple products.  Or we may want to consider moving our build tools into one or more separate accounts since the permissions needed and tasks done for building and deploying are completely separate from the running of the applications.  AWS has some recommendations regarding this, but I think we have bigger problems to solve in the short and medium-term future.  :smile:

For future reference and in case anyone is wondering how to do it, the cross-account access is configured in the ECR repo permissions page here:  https://us-west-1.console.aws.amazon.com/ecr/repositories/private/697549588058/mjml-server/_/permissions?region=us-west-1

**Review by LBarovic — APPROVED** [review]

**Comment by llostris** [comment]
@fendy-myeatclub Are they in CloudWatch? If so I, I think that would be fine – let's just document it somewhere :)

File: .github/workflows/build_and_release_ecr_image.yml
Comment (llostris) [reply .github/workflows/build_and_release_ecr_image.yml:51]: Thank you!!!

**Comment by fendy-myeatclub** [comment]
> @fendy-myeatclub Are they in CloudWatch? If so I, I think that would be fine – let's just document it somewhere :)

Correct. Will do.

**Pull request merged**
